cuqi/testproblem/_testproblem.py

```diff
@@ -1156,11 +1156,18 @@ class Deconvolution2D(BayesianProblem):
         | 'Mirror' - Mirrored boundary
         | 'Nearest' - Replicates last element of boundary
         
-    phantom : string
+    phantom : string or ndarray
         | The phantom (sharp image) that is convolved.
+        | If ndarray it should be a 2D array representing an image.
+        | The image will automatically be resized to fit the problem size.
+        | If string it should be any 2D phantom defined in cuqi.data.
+        | The string is lowercased and any hyphens are replaced 
+        | with underscores to match a method name in cuqi.data.
+        | Examples:
         | 'astronaut' - a photo of an astronaut.
         | 'camera' - a photo of a man with a camera.
         | 'cat' - a photo of a cat.
+        | 'cookie' - cartoon cookie.
```

Comment:
I have few comments here for your consideration.

1- Is phantom limited to square phantoms here? if so maybe this can be mentioned in the documentation when talking about the possibility of passing 2D array. If not, then do we need to update the property dim to be tuple? to be reasonable for non-square images.

2- The module cuqi.data has the following components  [satellite, astronaut, camera, cat, cookie, grains, shepp_logan, threephases, p_power, rgb2gray, imresize]. I think it will be nice to give the user a way to easily and immediately distinguish which are 1D phantoms, 2D phantoms and neither of these methods. Possible way to achieve that is to create 2 methods: `get_1D_phantom_names()` and `get_2D_phantom_names()` so the user can have the list (that could grow in the future) of phantoms. This also can be written in the documentation here: "To get the list of all 2D phantom, call cuqi.data.get_1D_phantom_names()"

3- if (2) above is addressed, maybe there is no need to mention "The string is lowercased and any hyphens are replaced with underscores to match a method name in cuqi.data." as it seems a bit technical and the user will have access to the phantom name list . Might say, phantom name is case insensitive.

Reply:
Added issue #278. Updated wording to reflect image should be square

Reply:
Thank you @nabriis for addressing this.